src/borg/remote.py

```diff
@@ -17,7 +17,8 @@
 from .helpers import bin_to_hex
 from .repository import Repository
 
-RPC_PROTOCOL_VERSION = 2
+RPC_PROTOCOL_VERSION = 3
+RPC_PROTOCOL_DOWNGRADABLE_VERSION = 2
```

Comment:
This might be crazy, but what if we have my version of the negotiate function first and then if that succeeds with `version >= 3` we go to `list_negotiate` which is basically your version but the client sends a list?